index.js

```diff
@@ -1,5 +1,6 @@
-const { fastifyRequestContextPlugin } = require('./lib/requestContextPlugin')
+const { fastifyRequestContextPlugin, requestContext } = require('./lib/requestContextPlugin')
 
 module.exports = {
   fastifyRequestContextPlugin,
+  requestContext,
```

Comment:
Wasn't this object already exposed as decorator to the fastify instance?

https://github.com/fastify/fastify-request-context/blob/1a7861b48cecc1a4acfd4488ac322111f1af5fbf/lib/requestContextPlugin.js#L12

It doesn't seem to be documented

Reply:
@Eomm It was, and it was even documented: `Request context (with methods get and set) is exposed by library itself, but is also available on fastify app instance as well as on req request instance.`

However, @puzpuzpuz made a good point that this introduces unnecessary complexity since some of the internal services may not necessarily have reference to `fastify` instance, so it might be convenient to be able to access it globally.

Reply:
@Eomm However, I still kept it on req and app, since app.[plugin] seems to be quite idiomatic for most plugins (`fastify.redis` etc), while `req.requestContext` feels like the most convenient way to use it from hooks. Hence I don't think that current situation when there are three ways to do same thing, is unnecessarily redundant.

Reply:
> It was, and it was even documented

Thanks, I was looking for "decorate" word or `fastify.requestContext`. I can add this example as improvement

>  is unnecessarily redundant.

Yeah, I read in the issue that the context would be a singleton in any case.
My fear is that having a singleton would be harder for the user to test it, for this I would suggest using the context as an input parameter instead to get it from a `require`. 
Bus as you said, if both the statement are documented, the user is free to choose what he/she prefers

Reply:
```
I can add this example as improvement
```

That would be most appreciated!